src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -188,6 +206,133 @@ def generate_retry_interval(retry_count: int) -> float:
+):
+    try:
+        # Does not support dynamic list for local.
+        from azure.mgmt.cognitiveservices import CognitiveServicesManagementClient
+        from promptflow.azure.operations._arm_connection_operations import \
+            ArmConnectionOperations, OpenURLFailedUserError
+    except ImportError:
+        return None
+
+    # For local, subscription_id is None. Does not support dynamic list for local.
+    if not subscription_id:
+        return None
+
+    try:
+        credential = _get_credential()
+        try:
+            # Currently, the param 'connection' is str, not AzureOpenAIConnection type.
+            conn = ArmConnectionOperations._build_connection_dict(
+                name=connection,
+                subscription_id=subscription_id,
+                resource_group_name=resource_group_name,
+                workspace_name=workspace_name,
+                credential=credential
+            )
+            resource_id = conn.get("value").get('resource_id', "")
+            if not resource_id:
+                return None
+            conn_sub, conn_rg, conn_account = _parse_resource_id(resource_id)
+        except OpenURLFailedUserError:
+            return None
+        except ListDeploymentsError as e:
+            raise e
+        except Exception as e:
+            msg = f"Parsing connection with exception: {e}"
+            raise ListDeploymentsError(msg=msg) from e
+
+        client = CognitiveServicesManagementClient(
+            credential=credential,
+            subscription_id=conn_sub,
+        )
+        return client.deployments.list(
+            resource_group_name=conn_rg,
+            account_name=conn_account,
+        )
+
+    except Exception as e:
+        if hasattr(e, 'status_code') and e.status_code == 403:
+            msg = f"Failed to list deployments due to permission issue: {e}"
+            raise ListDeploymentsError(msg=msg) from e
+        else:
+            msg = f"Failed to list deployments with exception: {e}"
+            raise ListDeploymentsError(msg=msg) from e
+
+
+def handle_unsupported_model_error(connection, deployment_name, model):
+    error_message = "LLM supports only text models. Please ensure you're using the correct tool for your model."
+    try:
+        if isinstance(connection, AzureOpenAIConnection):
+            subscription_id, resource_group, workspace_name = get_workspace_details()
+            if subscription_id and resource_group and workspace_name:
+                deployment_collection = list_deployment_connections(subscription_id, resource_group, workspace_name,
+                                                                    connection.name)
+                for item in deployment_collection:
+                    if deployment_name == item.name:
+                        if item.properties.model.version in [GPT4V_VERSION]:
+                            error_message = "LLM supports only text model for azure openai connection. " \
```

Comment:
tsg  = "Please kindly avoid using vision model in LLM tool because vision model cannot work with some chat api parameters. You can change to use tool 'Azure OpenAI GPT-4 Turbo with Vision' or 'OpenAI GPT-4V' for vision model."